GeneralsMD/Code/GameEngine/Include/Common/Money.h

```diff
@@ -105,6 +110,9 @@ class Money : public Snapshot
 
 	UnsignedInt m_money;	///< amount of money
 	Int m_playerIndex;	///< what is my player index?
+	UnsignedInt m_incomeBuckets[60];	///< circular buffer of 60 seconds for income tracking
+	UnsignedInt m_currentBucket;
+	UnsignedInt m_cashPerMinute;
```

Comment:
Helmut raised that this needs to be signed integer.

For example if Lotus is stealing but player is earning nothing, then it will be negative.